### ISSUE TITLE
XDP: how to use Redirect verdict

### ISSUE THREAD

**Issue by trinity-1686a** [redsift/redbpf#79]
Hi,
I wanted to build a simple XDP-based router using these crates as a test project, but I'm a little puzzled on how to use [XdpAction::Redirect](https://ingraind.org/api//redbpf_probes/xdp/enum.XdpAction.html#variant.Redirect).
Looking at some C examples over the net, it appears one is supposed to use [bpf_redirect](https://ingraind.org/api//redbpf_probes/helpers/gen/fn.bpf_redirect.html)/[bpf_redirect_map](https://ingraind.org/api//redbpf_probes/helpers/gen/fn.bpf_redirect_map.html) to get the right action number, however this returns a c_int (i32), which can't be cast to a XdpAction without transmuting it (which is safe due to repr(u32), but not convenient nor intuitive).
Am I correct, or is there a better way/an helper function I missed?

**Comment by rsdy** [comment]
I don't think we have the APIs fleshed out for this use case at the moment. I'll need to take a look into improving the XDP API also due to #78 , so if you have any concrete ideas or code examples you've tried, they would be welcome.
